tasks/projects.yml

```diff
@@ -78,22 +78,50 @@
   loop_control:
     label: "{{ item.name }}"
 
+- name: List OpenStack projects
+  shell: >
+    . {{ os_projects_venv }}/bin/activate &&
+    openstack
+    {% for auth_name, auth_value in os_projects_admin_auth.items() %}
+    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
+    {% endfor %}
+    {% if os_projects_cacert is defined %}
+    --os-cacert='{{ os_projects_cacert }}'
+    {% endif %}
+    {% if os_projects_cloud is defined %}
+    --os-cloud='{{ os_projects_cloud }}'
+    {% endif %}
+    --os-interface={{ os_projects_interface | default('public', true) }}
+    project list -f value -c Name
+  changed_when: False
+  register: projects_list
+
 - name: Ensure the project exists
-  os_project:
-    auth_type: "{{ os_projects_auth_type }}"
-    auth: "{{ os_projects_admin_auth }}"
-    cacert: "{{ os_projects_cacert | default(omit) }}"
-    cloud: "{{ os_projects_cloud | default(omit) }}"
-    interface: "{{ os_projects_interface | default(omit, true) }}"
-    name: "{{ item.name }}"
-    description: "{{ item.description }}"
-    domain_id: "{{ domain_is_id | ternary(item.project_domain, os_projects_domain_to_id[item.project_domain]) }}"
-    state: present
-    enabled: True
-    wait: yes
+  shell: >
+    . {{ os_projects_venv }}/bin/activate &&
+    openstack
+    {% for auth_name, auth_value in os_projects_admin_auth.items() %}
+    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
+    {% endfor %}
+    {% if os_projects_cacert is defined %}
+    --os-cacert='{{ os_projects_cacert }}'
+    {% endif %}
+    {% if os_projects_cloud is defined %}
+    --os-cloud='{{ os_projects_cloud }}'
+    {% endif %}
+    --os-interface={{ os_projects_interface | default('public', true) }}
+    project create {{ item.name }}
+    {% if item.description is defined %}
+    --description '{{ item.description }}'
+    {% endif %}
+    {% if item.project_domain is defined %}
+    --domain '{{ item.project_domain }}'
+    {% endif %}
+    {% if item.parent is defined %}
+    --parent '{{ item.parent }}'
+    {% endif %}
   with_items: "{{ os_projects }}"
-  vars:
-    domain_is_id: "{{ item.project_domain in os_projects_domain_to_id.values() }}"
+  when: item.name not in projects_list.stdout_lines
```

Comment:
We would lose the ability to update projects with this change

Reply:
Good point.

Reply:
Could use a separate `openstack project set` task?

Reply:
A simpler approach is to keep using `os_project` for updating project information such as description.

Note that it isn't possible to update the parent of a project, I will add a check for this in a follow-up.